chains/evm/calls/transactor/transact.go

```diff
@@ -0,0 +1,98 @@
+package transactor
+
+import (
+	"context"
+	"fmt"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/client"
+	"github.com/imdario/mergo"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog/log"
+)
+
+type TransactOptions struct {
+	GasLimit uint64
+	GasPrice *big.Int
+	Value    *big.Int
+}
+
+func DEFAULT_TRANSACTION_OPTIONS() TransactOptions {
+	return TransactOptions{
+		GasLimit: 2000000,
+		GasPrice: big.NewInt(0),
+		Value:    big.NewInt(0),
+	}
+}
+
+func FillTransactOptions(t TransactOptions) TransactOptions {
+	return MergeTransactionOptions(t, DEFAULT_TRANSACTION_OPTIONS())
+}
+
+func MergeTransactionOptions(primary TransactOptions, additional TransactOptions) TransactOptions {
+	if err := mergo.Merge(&primary, additional); err != nil {
+		log.Fatal().Msg("Unable to merge")
+		return TransactOptions{}
+	}
+	return primary
+}
+
+type Transactor interface {
+	Transact(to *common.Address, data []byte, opts TransactOptions) (*common.Hash, error)
+}
+
+type signAndSendTransactor struct {
+	TxFabric       client.TxFabric
+	gasPriceClient client.GasPricer
+	client         client.ClientDispatcher
+}
+
+func NewSignAndSendTransactor(txFabric client.TxFabric, gasPriceClient client.GasPricer, client client.ClientDispatcher) Transactor {
+	return &signAndSendTransactor{
+		TxFabric:       txFabric,
+		gasPriceClient: gasPriceClient,
+		client:         client,
+	}
+}
+
+func (t *signAndSendTransactor) Transact(to *common.Address, data []byte, opts TransactOptions) (*common.Hash, error) {
+	defer t.client.UnlockNonce()
+	t.client.LockNonce()
+	n, err := t.client.UnsafeNonce()
+	if err != nil {
+		return &common.Hash{}, nil
+	}
+	opts = FillTransactOptions(opts)
```

Comment:
So by doing this, I enable calling different contract calls by providing empty `transactor.TransactOptions{}` as a selector for default options. Still not 100% sold on this solution, what do you think @P1sar @mpetrun5  ?

Reply:
Looks fine for now! You can easly provide any opts from anywhere, that makes its easier to use thins function

Reply:
There is `MergeTransactionOptions` function that can merge any two `TransactOptions`. I will just make this function private as it is only here.

Reply:
There is `MergeTransactionOptions` function that can merge any two `TransactOptions`. I will just make this function private as it is only used here.